fix(curriculum): update debugging challenge (fcc vs browser console) (#36703)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/debugging/understanding-the-differences-between-the-freecodecamp-and-browser-console.english.md

The following challenge is meant to highlight the main difference between the freeCodeCamp console and your browser console.
When you run ordinary JavaScript, the browsers console will display your <code>console.log()</code> statements the exact number of times you requested.
The freeCodeCamp console will print your <code>console.log()</code> statements for each test of that challenge, as well as one more time for any function calls that you have in your code.
This lends itself to some interesting behavior and might trip you up in the beginning, because a logged value that you expect to see only once may print out many more times.
If you would like to see only your single output and not have to worry about running through the test cycles, you can use <code>console.clear()</code> and check the browsers console.
</section>

## Instructions
<section id='instructions'>
First, use <code>console.clear()</code> to clear the browser console. After that, use <code>console.log</code> to log the <code>output</code> variable.
</section>

## Tests
<section id='tests'>

```yml
tests:
  - text: Use <code>console.clear()</code> to clear the browser console.

>     testString: assert(code.match(/^(\s*console.clear\(\);?\s*)$/gm));

Reviewer: To allow for whites space can we use a helper function?  Also, if we are making the semi-colon optional, might as well get rid of it. I left them in, but do we really need the `^` and `$` here?
```suggestion
    testString: const noSpaces = code.replace(/\s/g, ''); assert(noSpaces.match(/^console.clear\(\)$/gm));
```

Author: I didn't really take a look at the tests, just changed what was there to check for the new variable name. No, we probably don't need the `^` or `?` or the `m`.

If you want to change your suggestion, I can commit them here.

Reviewer: I have edited both of my original suggestions.

  - text: Use <code>console.log()</code> to print the <code>output</code> variable.

>     testString: assert(code.match(/console\.log\(output\)/g));

Reviewer: To allow for white space, can we use a helper?
```suggestion
    testString: const noSpaces = code.replace(/\s/g, ''); assert(code.match(/console\.log\(output\)/));
```

Author: This should work here since we are logging variables. Don't you want it to be like this though...
```suggestion
assert(noSpaces.match(/console\.log\(output\)/));
```
`noSpaces.match` instead of `code.match` like the other suggestion?

Reviewer: Yeah, I meant to replace the `code` variable with `noSpaces`.  I will update my suggestion so you can commit it.

```js

> // Open your browser console

Reviewer 2: ```suggestion
// Open your browser console.
```

let output = "Get this to log once in the browser console and twice in the freeCodeCamp console";
// Use console.clear() on the next line to clear the browser console.

> // Use console.log() to print the output variable

Reviewer 2: ```suggestion
// Use console.log() to print the output variable.
```

> // Check the two consoles to see the difference. The freeCodeCamp console should have printed the variable twice, one for each test of this challenge. The browser console should only print the variable once becuase you cleared it first.

Reviewer 2: ```suggestion
// Check the two consoles to see the difference. The freeCodeCamp console should have printed the variable twice, once for each test of this challenge. The browser console should only print the variable once because you cleared it first.
```

```js

> // Open your browser console

Reviewer 2: ```suggestion
// Open your browser console.
```

let output = "Get this to log once in the browser console and twice in the freeCodeCamp console";
// Use console.clear() on the next line to clear the browser console.
console.clear();

> // Use console.log() to print the output variable

Reviewer 2: ```suggestion
// Use console.log() to print the output variable.
```
